two_factor/apps.py

```diff
@@ -10,3 +13,21 @@ def ready(self):
         if getattr(settings, 'TWO_FACTOR_PATCH_ADMIN', True):
             from .admin import patch_admin
             patch_admin()
+
+
+@receiver(setting_changed, dispatch_uid="two_factor.handle_setting_changed")
+def handle_setting_changed(sender, setting: str, value, **kwargs):
+    from .admin import patch_admin, unpatch_admin, __default_admin_site__
+
+    is_patched = __default_admin_site__ is not None
+
+    if setting == "TWO_FACTOR_PATCH_ADMIN":
+        if value is True and not is_patched:
+            patch_admin()
+        elif value is False and is_patched:
+            unpatch_admin()
+
+    elif setting == "TWO_FACTOR_FORCE_OTP_ADMIN":
+        if is_patched:
```

Comment:
you can probably put this "if" in the "elif statement" in order to combine them no?
`elif setting == "TWO_FACTOR_FORCE_OTP_ADMIN" and is_patched:`

Reply:
I decided to go with readability & symmetry with the above if-statement to make it clear we're only looking at two settings.